File: sora-editor/src/main/java/com/pranav/completion/KindDrawable.java
package com.pranav.completion;

import android.graphics.PixelFormat;
import android.graphics.drawable.Drawable;
import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.content.Context;
import android.content.res.Resources;

public class KindDrawable extends Drawable {
    
    private final Paint mPaint;
    private final Paint mTextPaint;
    
    private static Resources mResources;
    private final Kind mKind;
    
    public static void setResources(Resources res) {
        mResources = res;
    }
    
    public KindDrawable(Kind kind) {
        mKind = kind;
        
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setColor(kind.getColor());
        
        mTextPaint = new Paint();
        mTextPaint.setColor(0xffffffff);
        mTextPaint.setAntiAlias(true);
        mTextPaint.setTextSize(dp(14));
        mTextPaint.setTextAlign(Paint.Align.CENTER);
    }
    
    @Override
    public void draw(Canvas canvas) {
        float width = getBounds().right;
        float height = getBounds().bottom;

        canvas.drawRect(0, 0, width, height, mPaint);

        canvas.save();
        canvas.translate(width / 2f, height / 2f);
        float textCenter = (-(mTextPaint.descent() + mTextPaint.ascent()) / 2f);
        canvas.drawText(mKind.getValue(), 0, textCenter, mTextPaint);
        canvas.restore();
        
    }

    @Override
    public void setAlpha(int p1) {}

    @Override
    public void setColorFilter(ColorFilter p1) {}

    @Override
    public int getOpacity() {
        return PixelFormat.OPAQUE;
    }

    private float dp(int px) {
        return Math.round(mResources.getDisplayMetrics().density * px);
    }
}

File: sora-editor/src/main/java/com/pranav/completion/KindDrawable.kt
package com.pranav.completion

import android.graphics.PixelFormat
import android.graphics.drawable.Drawable
import android.graphics.Canvas
import android.graphics.ColorFilter
import android.graphics.Paint
import android.content.Context
import android.content.res.Resources

class KindDrawable(kind: Kind) : Drawable() {
    
    private val mPaint: Paint
    private val mTextPaint: Paint
    
    private var mResources: Resources?
    private val mKind: Kind

    companion object {
        @JvmStatic
        fun setResources(res: Resources) {
            mResources = res
        }
    }
    
    init {
        mKind = kind
        
        mPaint = Paint()
        mPaint.setAntiAlias(true)
        mPaint.setColor(kind.getColor())
        
        mTextPaint = Paint()
        mTextPaint.setColor(0xffffffff)
        mTextPaint.setAntiAlias(true)
        mTextPaint.setTextSize(dp(14))
        mTextPaint.setTextAlign(Paint.Align.CENTER)
    }
    
    override fun draw(canvas: Canvas) {
        var width: Float = getBounds().right
        var height: Float = getBounds().bottom

        canvas.drawRect(0, 0, width, height, mPaint)

        canvas.save()
        canvas.translate(width / 2f, height / 2f)
        var textCenter = (-(mTextPaint.descent() + mTextPaint.ascent()) / 2f)
        canvas.drawText(mKind.getValue(), 0, textCenter, mTextPaint)
        canvas.restore()
        
    }

    override fun setAlpha(p0: Int) {}

    override fun getOpacity(): Int {
        return PixelFormat.OPAQUE
    }

    private fun dp(px: Int): Float {
        return Math.round(mResources?.getDisplayMetrics().density * px)
    }
}